photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java

```diff
@@ -0,0 +1,146 @@
+            switch (entry.getKey()) {
+                case SIZE: {
+                    final float size = (float) entry.getValue();
+                    applyTextSize(textView, size);
+                }
+                break;
+
+                case COLOR: {
+                    final int color = (int) entry.getValue();
+                    applyTextColor(textView, color);
+                }
+                break;
+
+                case FONT_FAMILY: {
+                    final Typeface typeface = (Typeface) entry.getValue();
+                    applyFontFamily(textView, typeface);
+                }
+                break;
+
+                case GRAVITY: {
+                    final int gravity = (int) entry.getValue();
+                    applyGravity(textView, gravity);
+                }
+                break;
+
+                case BACKGROUND: {
+                    if (entry.getValue() instanceof Drawable) {
+                        final Drawable bg = (Drawable) entry.getValue();
+                        applyBackgroundDrawable(textView, bg);
+
+                    } else if (entry.getValue() instanceof Integer) {
+                        final int color = (Integer) entry.getValue();
+                        applyBackgroundColor(textView, color);
+                    }
+                }
+                break;
+
+                case TEXT_APPEARANCE: {
+                    if (entry.getValue() instanceof Integer) {
+                        final int styleAppearance = (Integer)entry.getValue();
+                        applyTextAppearance(textView, styleAppearance);
+                    }
+                }
+                break;
+            }
+        }
+    }
+
+    protected void applyTextSize(TextView textView, float size) {
+        textView.setTextSize(size);
+    }
+
+    protected void applyTextColor(TextView textView, int color) {
+        textView.setTextColor(color);
+    }
+
+    protected void applyFontFamily(TextView textView, Typeface typeface) {
+        textView.setTypeface(typeface);
+    }
+
+    protected void applyGravity(TextView textView, int gravity) {
+        textView.setGravity(gravity);
+    }
+
+    protected void applyBackgroundColor(TextView textView, int color) {
+        textView.setBackgroundColor(color);
+    }
+
+    protected void applyBackgroundDrawable(TextView textView, Drawable bg) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            textView.setBackground(bg);
+        } else {
+            textView.setBackgroundDrawable(bg);
+        }
+    }
+
+    protected void applyTextAppearance(TextView textView, int styleAppearance) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            textView.setTextAppearance(styleAppearance);
+        } else {
+            textView.setTextAppearance(textView.getContext(), styleAppearance);
+        }
+    }
+
+    protected enum TextStyle {
+        SIZE("TextSize"),
```

Comment:
Should go with TextStyle properties values to all small or all caps

Reply:
I don't get that, you mean the name of enum? if so, no, it should be like that  => [official doc](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)

If you mean the value, the properties don't need any format, it's a value wich could be any as you want, and I always prefer to write the values with camelcase